dojo-dungeon/src/dojo/monster/MyImp.java

```diff
@@ -18,8 +19,9 @@ public class MyImp {
    *
    * @param entity the entity to create
    * @param isInitialized if the entity was already initialized
```

Comment:
Zwei Fragen: 

- Sollte das nicht eine Eigenschaft der Entität sein? Warum wird das seperat übergeben?
- Ich habe das Gefühl, es geht gar nicht um den Parameter `entity`, sondern um den `MyImp`? Dann sollte das evtl. in der Klasse/Objekt von `MyImp`? gespeichert werden?

Reply:
> Zwei Fragen:
> 
> Sollte das nicht eine Eigenschaft der Entität sein? Warum wird das seperat übergeben?

> Ich habe das Gefühl, es geht gar nicht um den Parameter `entity`, sondern um den `MyImp`? Dann sollte das evtl. in der Klasse/Objekt von `MyImp`? gespeichert werden?

Das ist in der Tat etwas unschön. Ich ändere das in mehrere (statische) Methoden. Eine, um die Grundeigenschaften des Imps zu setzen (das habe ich hier etwas oberflächlich nur "Initialisierung" genannt), und mehrere andere, die von den Studis implementiert werden sollen, entsprechend der Java-Docs.

Reply:
Naja, vielleicht wolltest Du ja einen Builder bauen? Könnte auch sinnvoll sein. Oder eine Klasse mit statischen Methoden zum Konfigurieren des Imps. Oder eine richtige Klasse, die dann aber selbst eine Entity sein sollte (also ableiten sollte). 

Ich bin mir nicht sicher, wo Du hinwillst, such Dir das Passende aus :)

Reply:
> Naja, vielleicht wolltest Du ja einen Builder bauen? Könnte auch sinnvoll sein. Oder eine Klasse mit statischen Methoden zum Konfigurieren des Imps. Oder eine richtige Klasse, die dann aber selbst eine Entity sein sollte (also ableiten sollte).
> 
> Ich bin mir nicht sicher, wo Du hinwillst, such Dir das Passende aus :)

Ok, mache ich.

Im Moment ist es eher ein Sammelsurium aus unterschiedlich(st)en Ansätzen. 😀 Das ist nicht gut, stimmt.